src/net.cpp

```diff
@@ -1031,6 +1060,9 @@ void ThreadSocketHandler()
             BOOST_FOREACH(CNode* pnode, vNodesCopy)
                 pnode->Release();
         }
+
+        if (progress == 0)  // Nothing happened even though select did not block.  So slow us down.
+            MilliSleep(50);
```

Comment:
Select won't block because there is data available.  But we are not "harvesting" it because there is not enough room in the leaky bucket.

50msec is unfortunately completely arbitrary.